x-pack/legacy/plugins/infra/public/containers/logs/log_entries/index.ts

```diff
@@ -124,48 +138,92 @@ const useFetchEntriesEffect = (
   dispatch: Dispatch,
   props: LogEntriesProps
 ) => {
-  const { getLogEntriesAround, getLogEntriesBefore, getLogEntriesAfter } = useGraphQLQueries();
-
-  const [prevParams, cachePrevParams] = useState(props);
+  const [prevParams, cachePrevParams] = useState<LogEntriesProps | undefined>();
   const [startedStreaming, setStartedStreaming] = useState(false);
 
-  const runFetchNewEntriesRequest = async (override = {}) => {
+  const runFetchNewEntriesRequest = async (overrides: Partial<LogEntriesProps> = {}) => {
+    if (!props.startTimestamp || !props.endTimestamp) {
+      return;
+    }
+
     dispatch({ type: Action.FetchingNewEntries });
+
     try {
-      const payload = await getLogEntriesAround({
-        ...omit(props, 'jumpToTargetPosition'),
-        ...override,
-      });
+      const fetchArgs: LogEntriesRequest = {
+        sourceId: overrides.sourceId || props.sourceId,
+        startTimestamp: overrides.startTimestamp || props.startTimestamp,
+        endTimestamp: overrides.endTimestamp || props.endTimestamp,
+        query: overrides.filterQuery || props.filterQuery || undefined, // FIXME
+      };
+
+      if (props.timeKey) {
+        (fetchArgs as LogEntriesCenteredRequest).center = props.timeKey;
+      } else {
+        (fetchArgs as LogEntriesBeforeRequest).before = 'last';
+      }
+
+      const { data: payload } = await fetchLogEntries(fetchArgs);
       dispatch({ type: Action.ReceiveNewEntries, payload });
+
+      // Move position to the bottom if it's the first load.
+      // Do it in the next tick to allow the `dispatch` to fire
+      if (!props.timeKey && payload.bottomCursor) {
+        setTimeout(() => {
+          props.jumpToTargetPosition(payload.bottomCursor!);
+        });
+      }
     } catch (e) {
       dispatch({ type: Action.ErrorOnNewEntries });
     }
   };
 
   const runFetchMoreEntriesRequest = async (direction: ShouldFetchMoreEntries) => {
-    dispatch({ type: Action.FetchingMoreEntries });
+    if (!props.startTimestamp || !props.endTimestamp) {
+      return;
+    }
     const getEntriesBefore = direction === ShouldFetchMoreEntries.Before;
-    const timeKey = getEntriesBefore
-      ? state.entries[0].key
-      : state.entries[state.entries.length - 1].key;
-    const getMoreLogEntries = getEntriesBefore ? getLogEntriesBefore : getLogEntriesAfter;
+
+    // Control cursors are correct
+    if (getEntriesBefore && !state.topCursor) {
+      return;
+    } else if (!state.bottomCursor) {
+      return;
+    }
+
+    dispatch({ type: Action.FetchingMoreEntries });
+
     try {
-      const payload = await getMoreLogEntries({ ...props, timeKey });
+      const fetchArgs: LogEntriesRequest = {
+        sourceId: props.sourceId,
+        startTimestamp: props.startTimestamp,
+        endTimestamp: props.endTimestamp,
+        query: props.filterQuery || undefined, // FIXME
```

Comment:
in the API `query` can be `string` or `undefined`. `props.filterQuery` can be `string` or `null`. 

I tend to prefer `undefined` over `null` (I know the argument that `null` explicitly "no value", but IMO that can also be represented with `undefined` with less gotchas (i.e. `typeof null == 'object`). I don't know what's the official Elastic take on this, so I haven't fixed it yet).